### PR TITLE
Grouped coordinates in point `struct`s

### PR DESCRIPTION
**Pull request by Lehonti**
In `LivePreviewManager` and `AsyncEffectRenderer`

### CODE REVIEW

**Comment by cameronwhite** [comment]
It seems like these points are describing a rectangle, so maybe just go all the way to using a `RectangleI` instead?

**Comment by Lehonti** [comment]
Sure, good idea.

There is likely an off-by-one error in the original code (see line 321 of `AsyncEffectRenderer`, where `bounds` is assigned, which would end up with an overflow if an image is a single pixel wide or a single pixel high), so I changed that a little.

File: Pinta.Core/Classes/AsyncEffectRenderer.cs [hunk 1]
```diff
@@ -264,16 +262,19 @@ void RenderTile (int renderId, int threadId, int tileIndex)
 		// Update bounds to be shown on next expose.
 		lock (updated_lock) {
 			if (is_updated) {
-				updated_x1 = Math.Min (bounds.X, updated_x1);
-				updated_y1 = Math.Min (bounds.Y, updated_y1);
-				updated_x2 = Math.Max (bounds.X + bounds.Width, updated_x2);
-				updated_y2 = Math.Max (bounds.Y + bounds.Height, updated_y2);
+				updated_area = RectangleI.FromLTRB (
+					left: Math.Min (bounds.X, updated_area.X),
```
Comment (cameronwhite) [inline Pinta.Core/Classes/AsyncEffectRenderer.cs:266]: I think this is now just equivalent to `RectangleI.Union()`?

File: Pinta.Core/Classes/AsyncEffectRenderer.cs [hunk 2]
```diff
@@ -264,16 +262,19 @@ void RenderTile (int renderId, int threadId, int tileIndex)
...
+					right: Math.Max (bounds.X + bounds.Width - 1, updated_area.Right),
+					bottom: Math.Max (bounds.Y + bounds.Height - 1, updated_area.Bottom)
+				);
 			} else {
 				is_updated = true;
-				updated_x1 = bounds.X;
-				updated_y1 = bounds.Y;
-				updated_x2 = bounds.X + bounds.Width;
-				updated_y2 = bounds.Y + bounds.Height;
+				updated_area = new (
```
Comment (cameronwhite) [inline Pinta.Core/Classes/AsyncEffectRenderer.cs:273]: Can this just be `updated_area = bounds` ?

File: Pinta.Core/Classes/AsyncEffectRenderer.cs (hunk 1 above)
Comment (Lehonti) [reply Pinta.Core/Classes/AsyncEffectRenderer.cs]: Thanks, I didn't know this method already existed. I've updated the code accordingly

File: Pinta.Core/Classes/AsyncEffectRenderer.cs (hunk 2 above)
Comment (Lehonti) [reply Pinta.Core/Classes/AsyncEffectRenderer.cs]: Thanks for noticing this, I've now shortened the code according to your suggestion

**Pull request merged**